### PR TITLE
Fix fetch command

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Fix `aea fetch` command.

PR #1816 broke backward compatibility.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
n/a

### CODE REVIEW

File: aea/cli/fetch.py [hunk 1]
```diff
@@ -49,10 +49,15 @@
 def fetch(click_context, public_id, alias, local, remote):
     """Fetch Agent from Registry."""
     ctx = cast(Context, click_context.obj)
+    is_mixed = not local and not remote
+    ctx.set_config("is_local", local)
```
Comment (DavidMinarsch) [inline aea/cli/fetch.py:53]: above you did a bit stricter: `local and not remote`...I guess better as `fetch` is importable and therefore consistency checks not guaranteed

File: aea/cli/utils/package_utils.py [hunk 2]
```diff
@@ -283,58 +282,6 @@ def find_item_locally(
     return package_path, item_configuration
 
 
-def find_item_in_distribution(  # pylint: disable=unused-argument
```
Comment (DavidMinarsch) [inline aea/cli/utils/package_utils.py:286]: Please keep until we know for sure we don't need it anymore. Also keep in other places for now imo.

File: tests/test_cli/test_fetch.py [hunk 3]
```diff
@@ -125,6 +131,11 @@ def test_fetch_positive(self, *mocks):
             [*CLI_LOG_OPTION, "fetch", "--local", "author/name:0.1.0"],
             standalone_mode=False,
         )
+        self.runner.invoke(
+            cli,
+            [*CLI_LOG_OPTION, "fetch", "--remote", "author/name:0.1.0"],
+            standalone_mode=False,
+        )
```
Comment (DavidMinarsch) [inline tests/test_cli/test_fetch.py:138]: fetching repeatedly to same folder...should cause clashes

File: tests/test_cli/test_utils/test_utils.py [hunk 4]
```diff
@@ -320,48 +312,6 @@ def test_find_item_locally_cant_find(self, from_conftype_mock, *mocks):
         )
 
 
-class FindItemInDistributionTestCase(TestCase):
```
Comment (DavidMinarsch) [inline tests/test_cli/test_utils/test_utils.py:323]: See above

File: aea/cli/fetch.py (hunk 1 above)
Comment (marcofavorito) [reply aea/cli/fetch.py:53]: yes!

File: aea/cli/utils/package_utils.py (hunk 2 above)
Comment (marcofavorito) [reply aea/cli/utils/package_utils.py:286]: OK

File: tests/test_cli/test_fetch.py (hunk 3 above)
Comment (marcofavorito) [reply tests/test_cli/test_fetch.py:138]: OK, will split in 3 tests

File: tests/test_cli/test_utils/test_utils.py (hunk 4 above)
Comment (marcofavorito) [reply tests/test_cli/test_utils/test_utils.py:323]: OK

File: aea/cli/fetch.py (hunk 1 above)
Comment (marcofavorito) [reply aea/cli/fetch.py:53]: given the `registry_flag` decorator, it shouldn't happen that both `local` and `remote` are both true, but of course that could change for some reason, so being stricter gives more consistency

File: aea/cli/add.py [hunk 5]
```diff
@@ -167,3 +167,54 @@ def _add_item_deps(ctx: Context, item_type: str, item_config) -> None:
...
+    :return: the path to the found package.
+    """
+    try:
+        package_path = fetch_package(
+            item_type, public_id=item_public_id, cwd=ctx.cwd, dest=dest_path
+        )
+    except click.ClickException:
+        click.echo("Fetch from remote registry failed, trying locally...")
+        # the following might raise exception, but we don't catch it this time
+        package_path = find_item_locally_or_distributed(
```
Comment (marcofavorito) [inline aea/cli/add.py:217]: this is to ensure also in mixed mode we fetch distributed (only local)

File: tests/test_cli/test_fetch.py [hunk 6]
```diff
@@ -168,3 +195,83 @@ def test_fetch_agent_latest(self):
             "fetch", "--local", str(MY_FIRST_AEA_PUBLIC_ID.to_latest())
         )
         assert "my_first_aea" in os.listdir(self.t)
+
+
+class TestFetchAgentMixed(BaseAEATestCase):
```
Comment (marcofavorito) [inline tests/test_cli/test_fetch.py:200]: just moved from below

File: tests/test_cli/test_registry/test_fetch.py [hunk 7]
```diff
@@ -142,86 +134,3 @@ def teardown_class(cls):
             shutil.rmtree(cls.t)
         except (OSError, IOError):
             pass
-
-
-class TestFetchAgentMixed(BaseAEATestCase):
```
Comment (marcofavorito) [inline tests/test_cli/test_registry/test_fetch.py:147]: just moved to above

File: aea/cli/utils/package_utils.py [hunk 8]
```diff
@@ -27,8 +27,7 @@
 import click
 from jsonschema import ValidationError
 
-from aea import AEA_DIR
```
Comment (DavidMinarsch) [inline aea/cli/utils/package_utils.py:30]: import changed

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**